workflow/scripts/rule_zarr_stats.py

```
@@ -0,0 +1,274 @@
+
+import matplotlib.pyplot as plt
+import numpy
+import pandas as pd
+import sgkit
+from common_functions import make_filter_key
+from common_functions import sliding_window_density
+
+ds_dir = snakemake.input[0].replace(".vcf_done", "")
+ds = sgkit.load_dataset(ds_dir)
+ds = ds.sel(
+    samples=~ds[f"sample_{snakemake.wildcards.subset_name}_subset_mask"].values,
```

Ruff (0.14.5)

Undefined name `snakemake`

(F821)